packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -165,16 +168,25 @@ export function useGridColumns(columns: GridColumns, apiRef: GridApiRef): void {
     [forceUpdate, getColumnFromField, updateColumns],
   );
 
-  const moveColumn = React.useCallback(
+  const setColumnIndex = React.useCallback(
     (field: string, targetIndexPosition: number) => {
       logger.debug(`Moving column ${field} to index ${targetIndexPosition}`);
       const oldIndexPosition = gridState.columns.all.findIndex((col) => col === field);
 
+      apiRef.current.publishEvent(GRID_COLUMN_ORDER_CHANGED, {
+        field,
+        element: apiRef.current.getColumnHeaderElement(field),
+        colDef: getColumnFromField(field),
```

Comment:
better to use apiRef getColumnFromField to avoid the dependency on the function which allow to split this file more easily